services/QuillLMS/spec/models/sales_form_submission_spec.rb

```diff
@@ -11,48 +11,77 @@
     it { should validate_presence_of(:email) }
-    it { should validate_presence_of(:phone_number) }
-    it { should validate_presence_of(:zipcode) }
     it { should validate_presence_of(:collection_type) }
     it { should validate_presence_of(:school_premium_count_estimate) }
     it { should validate_presence_of(:teacher_premium_count_estimate) }
     it { should validate_presence_of(:student_premium_count_estimate) }
     it { should validate_presence_of(:submission_type) }
   end
 
-  context 'school name validation with blank district name' do
-    let(:sales_form_submission) { build(:sales_form_submission, district_name: '') }
+  context 'after_save, should perform this behavior on internal User records' do
+    let(:school) { create(:school)}
 
-    it { expect(sales_form_submission).to be_valid }
-    it { should validate_presence_of(:school_name) }
-    it { should validate_presence_of(:district_name) }
+    #before do
+    #  # For this context block, we don't actually care about testing Vitally
+    #  # behavior.  See below for those tests.
+    #  allow_any_instance_of(SalesFormSubmission).to receive(:send_opportunity_to_vitally)
+    #  allow_any_instance_of(SalesFormSubmission).to receive(:create_school_or_district_if_none_exist)
+    #  allow_any_instance_of(SalesFormSubmission).to receive(:vitally_user_create_data)
+    #  allow(api_double).to receive(:exists?).and_return(false)
+    #  allow(api_double).to receive(:create)
+    #end
+
+    it 'creates a new user record if a User does not already exist' do
+      expect { create(:sales_form_submission, school_name: school.name) }.to change(User, :count).by(1)
+    end
+
+    it 'creates a new user record with the email from the sales form submission' do
+      sales_form_submission = create(:sales_form_submission, school_name: school.name)
+      expect(User.find_by(email: sales_form_submission.email, role: User::SALES_CONTACT)).to be
+    end
+
+    it 'does not create a new user record if a User already exists' do
+      user = create(:user)
+      expect { create(:sales_form_submission, email: user.email, school_name: school.name) }.to change(User, :count).by(0)
+    end
   end
 
-  context 'district name validation with blank school name' do
-    let(:sales_form_submission) { build(:sales_form_submission, school_name: '') }
+  context 'after_save, it should enqueue Sidekiq jobs to sync data to Vitally and to send a Slack message' do
+    it 'should enqueue jobs after save' do
+      sales_form_submission = create(:sales_form_submission)
 
-    it { expect(sales_form_submission).to be_valid }
-    it { should validate_presence_of(:school_name) }
-    it { should validate_presence_of(:district_name) }
+      expect(SyncSalesFormSubmissionToVitallyWorker).to receive(:perform_async).with(sales_form_submission.id)
+      expect(SendSalesFormSubmissionToSlackWorker).to receive(:perform_async).with(sales_form_submission.id)
+
+      sales_form_submission.created_at = Time.now
```

Comment:
I'm not sure why the linter didn't catch this but we should use Time.current over Time.now.   Also, what's the reasoning behind updating the created_at here.